lib/Bridge/TolerantParser/ResolveAliasSuggestionCompletor.php

```diff
@@ -0,0 +1,150 @@
+use Microsoft\PhpParser\Parser;
+use Microsoft\PhpParser\ResolvedName;
+use Phpactor\Completion\Core\Completor;
+use Phpactor\Completion\Core\Suggestion;
+use Phpactor\Completion\Core\Util\OffsetHelper;
+use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\TextDocument;
+
+final class ResolveAliasSuggestionCompletor implements Completor
+{
+    /**
+     * @var Completor
+     */
+    private $decorated;
+
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    public function __construct(Completor $decorated, Parser $parser = null)
+    {
+        $this->decorated = $decorated;
+        $this->parser = $parser ?: new Parser();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function complete(TextDocument $source, ByteOffset $byteOffset): Generator
+    {
+        $importTable = $this->getClassImportTableAtPosition($source, $byteOffset);
+        $suggestions = $this->decorated->complete($source, $byteOffset);
+
+        foreach ($suggestions as $suggestion) {
+            $resolvedSuggestions = $this->resolveAliasSuggestions($importTable, $suggestion);
+
+            // Trick to avoid any BC break when converting to an array
+            // https://www.php.net/manual/fr/language.generators.syntax.php#control-structures.yield.from
+            foreach ($resolvedSuggestions as $resolvedSuggestion) {
+                yield $resolvedSuggestion;
+            }
+        }
+
+        return $suggestions->getReturn();
+    }
+
+    private function truncateSource(string $source, int $byteOffset): string
+    {
+        // truncate source at byte offset - we don't want the rest of the source
+        // file contaminating the completion (for example `$foo($<>\n    $bar =
+        // ` will evaluate the Variable node as an expression node with a
+        // double variable `$\n    $bar = `
+        $truncatedSource = substr($source, 0, $byteOffset);
+
+        // determine the last non-whitespace _character_ offset
+        $characterOffset = OffsetHelper::lastNonWhitespaceCharacterOffset($truncatedSource);
+
+        // truncate the source at the character offset
+        $truncatedSource = mb_substr($source, 0, $characterOffset);
+
+        return $truncatedSource;
+    }
+
+    /**
+     * Add suggestions when a class is already imported with an alias or when a relative name is abailable.
+     *
+     * Will update the suggestion to remove the import_name option if already imported.
+     * Will add a suggestion if the class is imported under an alias.
+     * Will add a suggestion if part of the namespace is imported (i.e. ORM\Column is a relative name).
+     *
+     * @param ResolvedName[] $importTable
+     *
+     * @return Suggestion[]
+     */
+    private function resolveAliasSuggestions(array $importTable, Suggestion $suggestion): array
+    {
+        if (Suggestion::TYPE_CLASS !== $suggestion->type()) {
+            return [$suggestion];
+        }
+
+        $suggestionFqcn = $suggestion->classImport();
```

Comment:
you should use `nameImport` instead (`classImport` is deprecated) maybe this also helps it work with functions?

Reply:
I realized it when doing the feature but forgot to check what I had already done, I will fix it.
I'm not sure to understand the question about the functions, for now I wanted to keep it simple.
But indeed I had in mind to make it work with functions as well at some point, that's way I don't mention "class" in the name of the decorator.